scripts/error-extractor/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+var path = require('path');
+var traverse = require('babel-traverse').default;
+
+var evalToString = require('./evalToString');
+var getFilePaths = require('./getFilePaths');
+var invertObject = require('./invertObject');
+var parseJSFile = require('./parseJSFile');
+var writeJSON = require('./writeJSON');
+
+var outputFileName = 'ErrorMap.json';
+var outputFilePath = path.join(__dirname, '../../', outputFileName);
+
+var existingErrorMap;
+var currentID;
+
+try {
+  existingErrorMap = require(outputFilePath);
+} catch (e) {
+  existingErrorMap = {};
+}
+
+// Now the error map is in the format of `{ 0: 'MUCH ERROR', 1: 'SUCH WRONG' }`
+
+var currentIDNumbers = Object.keys(existingErrorMap).map(function(str) {
+  return parseInt(str, 10);
+});
+
+if (currentIDNumbers.length === 0) { // Map is empty
+  currentID = 0;
+} else {
+  currentID = Math.max.apply(null, currentIDNumbers) + 1;
+}
+
+// Here we invert the map object in memory for faster error code lookup
+existingErrorMap = invertObject(existingErrorMap);
+
+var filePaths = getFilePaths();
+
+// Let's sort the files using their (module) names to make the result more stable
+filePaths.sort(function(path1/* : string */, path2/* : string */)/* : number */ {
```

Comment:
This complexity probably isn't necessary, especially because in most cases it won't matter since we're just adding one or two new errors (after the initial generation). How about we just `.sort()` with the full file path? Then you don't need a comparator.